packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/modals/JoinZoomNowModal.tsx

```diff
@@ -0,0 +1,79 @@
+import Modal from 'antd/lib/modal/Modal'
+import JoinZoomButton from '../JoinZoomButton'
+import ReQueuingButton from '../ReQueuingButton'
+import { Popconfirm } from 'antd'
+import { InfoCircleFilled } from '@ant-design/icons'
+import { useEffect, useState } from 'react'
+
+type JoinZoomNowModalProps = {
+  open: boolean
+  zoomLink?: string
+  taName?: string
+  notes?: string
+  onJoin: () => void
+  setRequeuing: () => Promise<void>
+}
+
+const JoinZoomNowModal: React.FC<JoinZoomNowModalProps> = ({
+  open,
+  zoomLink,
+  taName,
+  notes,
+  onJoin,
+  setRequeuing,
+}) => {
+  const [isLoadingRequeue, setIsLoadingRequeue] = useState(false)
+  const [isClosable, setIsClosable] = useState(false)
+  const [localOpen, setLocalOpen] = useState(true)
+
+  useEffect(() => {
+    setIsClosable(false)
+    if (open) {
+      setLocalOpen(true)
+      setTimeout(() => {
+        setIsClosable(true)
+      }, 10000)
+    }
+  }, [open])
+
+  return (
+    <Modal
+      open={isClosable ? localOpen : open}
+      footer={[]}
+      style={{ top: '30%' }}
+      closable={isClosable}
+      onCancel={() => setLocalOpen(false)}
```

Comment:
Okay I found a very interesting bug with this! 
Steps to reproduce:
- Start getting helped
- Before the 10s is over, click somewhere on the gray-ed out background
- Notice that once the 10s is up, the modal closes automatically
![image](https://github.com/user-attachments/assets/1bca0e46-0402-415f-9a32-417da4816ee2)
This is pretty interesting, and I guess it's because when you click on the grayed out background, it will run this onCancel which will setLocalOpen to False, meaning that once isClosable is true, it will  immediately set open to false. Luckily, I think all that's require to fix this is by adding a check to make sure isClosable is true before setting setLocalOpen to false.

Reply:
I just used another prop: "maskClosable" which just disables the ability to close the modal by clicking off entirely which I think is much better since there is no way to bring the modal back. So having a close button be the only button that can close the modal I think makes it most clear and prevents more accidental "misclicks".